packages/action/index.js

```diff
@@ -22,32 +22,78 @@ const core = require('@actions/core');
+    }
+
+   
+    let buildmessageStr = process.env.GITHUB_WORKFLOW + ' ' + process.env.GITHUB_RUN_NUMBER;
+    if(core.getInput('tag') && core.getInput('tag') !== "None"){
+      buildmessageStr += ' (' + core.getInput('tag') + ')';
+    }
+
 
     const metaData = {
-      github: JSON.parse(core.getInput('github')),
-      os: JSON.parse(core.getInput('os')),
-      matrix: JSON.parse(core.getInput('matrix')),
+      repoId: encodeURIComponent(process.env.GITHUB_REPOSITORY), //resanitized server side
+      organization: process.env.GITHUB_REPOSITORY.substr(0,process.env.GITHUB_REPOSITORY.indexOf("/")),
+      timestamp: new Date(),
+      url: process.env.GITHUB_SERVER_URL + '/' + process.env.GITHUB_REPOSITORY,
+      environment:{
+        os: core.getInput('os'),
+        tag: core.getInput('tag'),
+        matrix :JSON.parse(core.getInput('matrix')),
+        ref: process.env.GITHUB_REF
+      },
+      buildId: process.env.GITHUB_RUN_ID,
+      sha: process.env.GITHUB_SHA,
+      name: process.env.GITHUB_REPOSITORY.substr(1+process.env.GITHUB_REPOSITORY.indexOf("/")),
+      description: core.getInput('repo-description'),
+      buildmessage: buildmessageStr,
+      token: core.getInput('repo-token')
     };
-    const fileType = core.getInput('logtype');
-    console.log('reading from: ' + core.getInput('filepath'));
-    const data = fs.readFileSync(
-        core.getInput('filepath'), 'utf8');
 
+    metaData.matrix = JSON.stringify(metaData.environment.matrix, Object.keys(metaData.environment.matrix).sort()); //consistancy 
+    
+    metaData.environment.ref = metaData.environment.ref.replace('refs/', '');
+    metaData.environment.ref = metaData.environment.ref.replace('heads/', '');
+
+    const fileType = core.getInput('log-type');
+    
+    if (!fs.existsSync(core.getInput('file-path'))){
+      core.warning("Could not find a test log file located at " + core.getInput('file-path'));
+      core.warning("Make you are saving a test log before running this action, and that it is saved to the file-path arguement");
+      return;
+    }
+    const data = fs.readFileSync(
+        core.getInput('file-path'), 'utf8');
     const sendMe = {type: fileType, data: data, metadata: metaData};
-    const endpoint = core.getInput('endpoint') || 'https://flaky-dashboard.web.app/api/build'
+    const endpoint = core.getInput('endpoint') + '/api/build/gh/v1'
+    console.log("Beginning Upload of data...")
     const outcome = await fetch(endpoint, {
       method: 'POST',
       body: JSON.stringify(sendMe),
       headers: {'Content-Type': 'application/json'},
     });
     const outcomeText = await outcome.text();
-    console.log('\nSERVER RESPONSE:');
-    console.log(outcomeText);
+    const outcomeAsJSON = JSON.parse(outcomeText);
+    if(outcomeAsJSON.error){
+      core.warning("Upload Failed - Status " + outcome.status);
```

Comment:
if we get an error, we should probably:

```
process.exitCode = 1
```

☝️ this will mean that the action indicates a failure.

Reply:
Okay sounds good. I initially avoided this because it could be annoying if your entire action gets cancelled because we are having server issues.